dev_guide/builds.adoc

```diff
@@ -519,9 +519,21 @@ customStrategy:
 ----
 ====
 
+[[build-inputs]]
+== Build Inputs
+
+There are several ways to provide content for the build to operate on:  
+
+* Git repositories
+* Inline Dockerfile definitions
+* Binary inputs
+* Content extracted from existing images
+
+These inputs can also be combined in a single build.  When the build is run, a working directory is constructed and all input content is placed in the working directory (e.g. the input git repository is cloned into the working directory, files specified from input images are copied into the working directory using the target path).  Next the build process will `*cd*` into the `*contextDir*` if one is defined.  Finally the content from the current directory will be provided to the build process for reference by the `*Dockerfile*`, `*assemble*` script, or custom builder logic.
```

Comment:
> These inputs can also be combined in a single build.

It might be useful to explain how they can be combined and in what order. First you can have either git or binary, then on top of that image source and finally Dockerfile. 

> the input git repository is cloned into the working directory, files specified from input images are copied into the working directory using the target path

It's probably also worth noting that while binary, git and image source are placed at the root of the working directory, Dockerfile's are actually placed wherever your contextDir points.

We could also add a warning that states that if the destination directory of your image source falls outside of the context directory, then no part of your image source will be used in the build.